ccl/backupccl: TestBackupRestoreSystemJobs failed

ccl/backupccl.TestBackupRestoreSystemJobs [failed](https://tanzanite.cluster.engflow.com/invocations/default/8d5b8c67-35b6-4168-965f-87485669da8d?testReportRun=1&testReportShard=30&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [ff291e3ee70eedac50b985860502d2f0b0867003](https://github.com/cockroachdb/cockroach/commits/ff291e3ee70eedac50b985860502d2f0b0867003):


```
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    jobs_verification.go:169: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/jobutils/jobs_verification.go:210
        	            				github.com/cockroachdb/cockroach/pkg/testutils/jobutils/jobs_verification.go:247
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backup_test.go:1261
        	Error:      	Not equal: 
        	            	expected: jobs.Record{JobID:0, Description:"RESTORE TABLE bank FROM 'nodelocal://1/foo/full?AWS_SESSION_TOKEN=redacted', 'nodelocal://1/foo/inc?AWS_SESSION_TOKEN=redacted' WITH OPTIONS (into_db = 'restoredb')", Statements:[]string(nil), Username:"root", DescriptorIDs:descpb.IDs{0x6c, 0x6d}, Details:jobspb.Details(nil), Progress:jobspb.ProgressDetails(nil), RunningStatus:"", NonCancelable:false, CreatedBy:(*jobs.CreatedByInfo)(nil), MaximumPTSAge:0}
        	            	actual  : jobs.Record{JobID:0, Description:"RESTORE TABLE bank FROM 'nodelocal://1/foo/full?AWS_SESSION_TOKEN=redacted', 'nodelocal://1/foo/inc?AWS_SESSION_TOKEN=redacted' WITH OPTIONS (into_db = 'restoredb')", Statements:[]string(nil), Username:"root", DescriptorIDs:descpb.IDs{0x6c, 0x6e}, Details:jobspb.Details(nil), Progress:jobspb.ProgressDetails(nil), RunningStatus:"", NonCancelable:false, CreatedBy:(*jobs.CreatedByInfo)(nil), MaximumPTSAge:0}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -9,3 +9,3 @@
        	            	   (catid.DescID) 108,
        	            	-  (catid.DescID) 109
        	            	+  (catid.DescID) 110
        	            	  },
        	Test:       	TestBackupRestoreSystemJobs
    backup_test.go:1272: job 0 did not match:
        101, POLL JOBS STATS, POLL JOBS STATS, , root, {}, running, NULL, 2023-12-21 09:50:55.187331 +0000 UTC, 2023-12-21 09:50:59.37039 +0000 UTC, NULL, 2023-12-21 09:50:59.370428 +0000 UTC, 0, NULL, , 1, 5964507710310202425, 2023-12-21 09:50:59.370398 +0000 UTC, 2023-12-21 09:50:59.420398 +0000 UTC, 1, NULL, NULL
        102, AUTO CONFIG RUNNER, applies automatic configuration, , root, {}, running, NULL, 2023-12-21 09:51:02.460513 +0000 UTC, 2023-12-21 09:51:04.965479 +0000 UTC, NULL, 2023-12-21 09:51:04.965506 +0000 UTC, 0, NULL, , 1, 3184116534759806332, 2023-12-21 09:51:04.965484 +0000 UTC, 2023-12-21 09:51:05.015484 +0000 UTC, 1, NULL, NULL
        103, AUTO UPDATE SQL ACTIVITY, sql activity job, , node, {}, running, NULL, 2023-12-21 09:51:11.802376 +0000 UTC, 2023-12-21 09:51:15.477911 +0000 UTC, NULL, 2023-12-21 09:51:15.477952 +0000 UTC, 0, NULL, , 1, 1079945888252842369, 2023-12-21 09:51:15.477916 +0000 UTC, 2023-12-21 09:51:15.527916 +0000 UTC, 1, NULL, NULL
        104, MVCC STATISTICS UPDATE, mvcc statistics update job, , node, {}, running, NULL, 2023-12-21 09:51:19.475461 +0000 UTC, 2023-12-21 09:51:23.026148 +0000 UTC, NULL, 2023-12-21 09:51:23.026173 +0000 UTC, 0, NULL, , 1, 811509347916613252, 2023-12-21 09:51:23.026152 +0000 UTC, 2023-12-21 09:51:23.076152 +0000 UTC, 1, NULL, NULL
        927602429776494595, AUTO SPAN CONFIG RECONCILIATION, reconciling span configurations, , node, {}, running, NULL, 2023-12-21 09:49:56.196575 +0000 UTC, 2023-12-21 09:49:59.736149 +0000 UTC, NULL, 2023-12-21 09:53:18.968167 +0000 UTC, 0, NULL, , 3, 5587821987182715805, 2023-12-21 09:49:59.736153 +0000 UTC, 2023-12-21 09:49:59.786153 +0000 UTC, 1, NULL, NULL
        927602432461078531, MIGRATION, Upgrade to 0.0-upgrading-step-002: "add users and roles", , root, {}, succeeded, NULL, 2023-12-21 09:49:57.015723 +0000 UTC, 2023-12-21 09:50:00.297243 +0000 UTC, 2023-12-21 09:50:06.811596 +0000 UTC, 2023-12-21 09:50:06.811616 +0000 UTC, 1, NULL, , 3, 8231888046685401126, 2023-12-21 09:50:00.297247 +0000 UTC, 2023-12-21 09:50:00.347247 +0000 UTC, 1, NULL, NULL
        927602470582288387, MIGRATION, Upgrade to 0.0-upgrading-step-004: "enable diagnostics reporting", , root, {}, succeeded, NULL, 2023-12-21 09:50:08.649393 +0000 UTC, 2023-12-21 09:50:10.073488 +0000 UTC, 2023-12-21 09:50:12.769559 +0000 UTC, 2023-12-21 09:50:12.769605 +0000 UTC, 1, NULL, , 3, 8045823585330695963, 2023-12-21 09:50:10.073492 +0000 UTC, 2023-12-21 09:50:10.123492 +0000 UTC, 1, NULL, NULL
        927602489400721411, MIGRATION, Upgrade to 0.0-upgrading-step-008: "initialize the cluster.secret setting", , root, {}, succeeded, NULL, 2023-12-21 09:50:14.392406 +0000 UTC, 2023-12-21 09:50:16.490116 +0000 UTC, 2023-12-21 09:50:18.412265 +0000 UTC, 2023-12-21 09:50:18.412285 +0000 UTC, 1, NULL, , 3, 8781526757692588773, 2023-12-21 09:50:16.490121 +0000 UTC, 2023-12-21 09:50:16.540121 +0000 UTC, 1, NULL, NULL
        927602510728691715, MIGRATION, Upgrade to 0.0-upgrading-step-010: "update system.locations with default location data", , root, {}, succeeded, NULL, 2023-12-21 09:50:20.901122 +0000 UTC, 2023-12-21 09:50:23.621074 +0000 UTC, 2023-12-21 09:50:26.563383 +0000 UTC, 2023-12-21 09:50:26.563404 +0000 UTC, 1, NULL, , 3, 1838953349764589619, 2023-12-21 09:50:23.621079 +0000 UTC, 2023-12-21 09:50:23.671079 +0000 UTC, 1, NULL, NULL
        927602534743113731, MIGRATION, Upgrade to 0.0-upgrading-step-012: "create default databases", , root, {}, succeeded, NULL, 2023-12-21 09:50:28.22972 +0000 UTC, 2023-12-21 09:50:30.147982 +0000 UTC, 2023-12-21 09:50:34.009434 +0000 UTC, 2023-12-21 09:50:34.009457 +0000 UTC, 1, NULL, , 3, 3026148338674356173, 2023-12-21 09:50:30.147987 +0000 UTC, 2023-12-21 09:50:30.197987 +0000 UTC, 1, NULL, NULL
        927602561100513283, MIGRATION, Upgrade to 1000022.1-upgrading-to-1000022.2-step-042: "add default SQL schema telemetry schedule", , root, {}, succeeded, NULL, 2023-12-21 09:50:36.273371 +0000 UTC, 2023-12-21 09:50:38.686434 +0000 UTC, 2023-12-21 09:50:42.344909 +0000 UTC, 2023-12-21 09:50:42.344934 +0000 UTC, 1, NULL, , 3, 9177369785962411606, 2023-12-21 09:50:38.686439 +0000 UTC, 2023-12-21 09:50:38.736439 +0000 UTC, 1, NULL, NULL
        927602595845668865, MIGRATION, Upgrade to 1000022.2-upgrading-to-1000023.1-step-038: "create jobs metrics polling job", , root, {}, succeeded, NULL, 2023-12-21 09:50:46.876743 +0000 UTC, 2023-12-21 09:50:50.712888 +0000 UTC, 2023-12-21 09:50:56.351196 +0000 UTC, 2023-12-21 09:50:56.351215 +0000 UTC, 1, NULL, , 1, 8564952996569724262, 2023-12-21 09:50:50.712893 +0000 UTC, 2023-12-21 09:50:50.762893 +0000 UTC, 1, NULL, NULL
        927602634161160193, AUTO CREATE STATS, Table statistics refresh for system.public.statement_diagnostics, CREATE STATISTICS __auto__ FROM [36] WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-30s', root, {}, succeeded, NULL, 2023-12-21 09:50:58.569722 +0000 UTC, 2023-12-21 09:51:00.364497 +0000 UTC, 2023-12-21 09:51:09.298246 +0000 UTC, 2023-12-21 09:51:09.298383 +0000 UTC, 1, NULL, , 1, 4431961800208702128, 2023-12-21 09:51:00.364502 +0000 UTC, 2023-12-21 09:51:00.414502 +0000 UTC, 1, NULL, NULL
        927602634435624961, MIGRATION, Upgrade to 1000022.2-upgrading-to-1000023.1-step-090: "create auto config runner job", , root, {}, succeeded, NULL, 2023-12-21 09:50:58.653461 +0000 UTC, 2023-12-21 09:51:01.438779 +0000 UTC, 2023-12-21 09:51:03.938522 +0000 UTC, 2023-12-21 09:51:03.938544 +0000 UTC, 1, NULL, , 1, 8404922479823405385, 2023-12-21 09:51:01.438784 +0000 UTC, 2023-12-21 09:51:01.488784 +0000 UTC, 1, NULL, NULL
        927602660877729793, MIGRATION, Upgrade to 1000022.2-upgrading-to-1000023.1-step-102: "create sql activity updater job", , root, {}, succeeded, NULL, 2023-12-21 09:51:06.722959 +0000 UTC, 2023-12-21 09:51:09.440979 +0000 UTC, 2023-12-21 09:51:12.493714 +0000 UTC, 2023-12-21 09:51:12.493741 +0000 UTC, 1, NULL, , 1, 97731722500570010, 2023-12-21 09:51:09.44099 +0000 UTC, 2023-12-21 09:51:09.49099 +0000 UTC, 1, NULL, NULL
        927602687038750721, MIGRATION, Upgrade to 1000023.1-upgrading-to-1000023.2-step-030: "create system.mvcc_statistics table and job", , root, {}, succeeded, NULL, 2023-12-21 09:51:14.706665 +0000 UTC, 2023-12-21 09:51:18.064846 +0000 UTC, 2023-12-21 09:51:20.90468 +0000 UTC, 2023-12-21 09:51:20.904701 +0000 UTC, 1, NULL, , 1, 7286101648837484742, 2023-12-21 09:51:18.06485 +0000 UTC, 2023-12-21 09:51:18.11485 +0000 UTC, 1, NULL, NULL
        927602699311874049, AUTO CREATE STATS, Table statistics refresh for system.public.database_role_settings, CREATE STATISTICS __auto__ FROM [44] WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-30s', root, {}, succeeded, NULL, 2023-12-21 09:51:18.452122 +0000 UTC, 2023-12-21 09:51:19.666544 +0000 UTC, 2023-12-21 09:51:30.00676 +0000 UTC, 2023-12-21 09:51:30.00686 +0000 UTC, 1, NULL, , 1, 5229249054794641437, 2023-12-21 09:51:19.666549 +0000 UTC, 2023-12-21 09:51:19.716549 +0000 UTC, 1, NULL, NULL
        927602747176747009, AUTO CREATE STATS, Table statistics refresh for system.public.mvcc_statistics, CREATE STATISTICS __auto__ FROM [60] WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-30s', root, {}, succeeded, NULL, 2023-12-21 09:51:33.059324 +0000 UTC, 2023-12-21 09:51:34.929926 +0000 UTC, 2023-12-21 09:51:46.693912 +0000 UTC, 2023-12-21 09:51:46.694019 +0000 UTC, 1, NULL, , 1, 6048149663021003547, 2023-12-21 09:51:34.929931 +0000 UTC, 2023-12-21 09:51:34.979931 +0000 UTC, 1, NULL, NULL
        927602819589668865, AUTO CREATE STATS, Table statistics refresh for system.public.role_id_seq, CREATE STATISTICS __auto__ FROM [48] WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-30s', root, {}, succeeded, NULL, 2023-12-21 09:51:55.15799 +0000 UTC, 2023-12-21 09:51:56.15702 +0000 UTC, 2023-12-21 09:52:04.771346 +0000 UTC, 2023-12-21 09:52:04.771411 +0000 UTC, 1, NULL, , 1, 6557469211151466199, 2023-12-21 09:51:56.157026 +0000 UTC, 2023-12-21 09:51:56.207026 +0000 UTC, 1, NULL, NULL
        927602824113061890, AUTO CREATE STATS, Table statistics refresh for system.public.settings, CREATE STATISTICS __auto__ FROM [6] WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-30s', root, {}, succeeded, NULL, 2023-12-21 09:51:56.538425 +0000 UTC, 2023-12-21 09:52:02.388257 +0000 UTC, 2023-12-21 09:52:43.187293 +0000 UTC, 2023-12-21 09:52:43.187387 +0000 UTC, 1, NULL, , 2, 6401396193265193433, 2023-12-21 09:52:02.388262 +0000 UTC, 2023-12-21 09:52:02.438262 +0000 UTC, 1, NULL, NULL
        927602843720286209, BACKUP, BACKUP DATABASE data TO 'nodelocal://1/foo/full?AWS_SESSION_TOKEN=redacted', , root, {104,105,106}, succeeded, NULL, 2023-12-21 09:52:02.099029 +0000 UTC, 2023-12-21 09:52:04.684502 +0000 UTC, 2023-12-21 09:52:21.878604 +0000 UTC, 2023-12-21 09:52:21.878749 +0000 UTC, 1, NULL, , 1, 2507878558157886520, 2023-12-21 09:52:04.684507 +0000 UTC, 2023-12-21 09:52:04.734507 +0000 UTC, 1, NULL, NULL
        927602939440070657, BACKUP, BACKUP TABLE bank TO 'nodelocal://1/foo/inc?AWS_SESSION_TOKEN=redacted' INCREMENTAL FROM 'nodelocal://1/foo/full?AWS_SESSION_TOKEN=redacted', , root, {104,105,106}, succeeded, NULL, 2023-12-21 09:52:26.327532 +0000 UTC, 2023-12-21 09:52:36.155892 +0000 UTC, 2023-12-21 09:52:49.75214 +0000 UTC, 2023-12-21 09:52:49.752308 +0000 UTC, 1, NULL, , 1, 7918207521708799184, 2023-12-21 09:52:36.155899 +0000 UTC, 2023-12-21 09:52:36.205899 +0000 UTC, 1, NULL, NULL
        927602990149009409, AUTO CREATE STATS, Table statistics refresh for system.public.replication_critical_localities, CREATE STATISTICS __auto__ FROM [26] WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-30s', root, {}, succeeded, NULL, 2023-12-21 09:52:47.208592 +0000 UTC, 2023-12-21 09:52:49.588354 +0000 UTC, 2023-12-21 09:53:01.761237 +0000 UTC, 2023-12-21 09:53:01.761336 +0000 UTC, 1, NULL, , 1, 7465902073349299697, 2023-12-21 09:52:49.588359 +0000 UTC, 2023-12-21 09:52:49.638359 +0000 UTC, 1, NULL, NULL
        927603044595204097, RESTORE, RESTORE TABLE bank FROM 'nodelocal://1/foo/full?AWS_SESSION_TOKEN=redacted', 'nodelocal://1/foo/inc?AWS_SESSION_TOKEN=redacted' WITH OPTIONS (into_db = 'restoredb'), , root, {108,110}, succeeded, NULL, 2023-12-21 09:53:00.486241 +0000 UTC, 2023-12-21 09:53:05.643231 +0000 UTC, 2023-12-21 09:53:17.952416 +0000 UTC, 2023-12-21 09:53:17.952526 +0000 UTC, 1, NULL, , 1, 7233456896142124273, 2023-12-21 09:53:05.643237 +0000 UTC, 2023-12-21 09:53:05.693237 +0000 UTC, 1, NULL, NULL
        927603060624982017, AUTO CREATE STATS, Table statistics refresh for system.public.zones, CREATE STATISTICS __auto__ FROM [5] WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-30s', root, {}, succeeded, NULL, 2023-12-21 09:53:08.716134 +0000 UTC, 2023-12-21 09:53:10.392167 +0000 UTC, 2023-12-21 09:53:19.579868 +0000 UTC, 2023-12-21 09:53:19.579953 +0000 UTC, 1, NULL, , 1, 1054690723048879126, 2023-12-21 09:53:10.392172 +0000 UTC, 2023-12-21 09:53:10.442172 +0000 UTC, 1, NULL, NULL
        927603067839184898, AUTO CREATE STATS, Table statistics refresh for system.public.span_count, CREATE STATISTICS __auto__ FROM [50] WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-30s', root, {}, running, NULL, 2023-12-21 09:53:10.917729 +0000 UTC, 2023-12-21 09:53:15.972992 +0000 UTC, NULL, 2023-12-21 09:53:15.976465 +0000 UTC, 0, NULL, , 2, 7778604223724767825, 2023-12-21 09:53:15.973002 +0000 UTC, 2023-12-21 09:53:16.023002 +0000 UTC, 1, NULL, NULL
    testutils.go:289: no Invalid Descriptors
    panic.go:541: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupRestoreSystemJobs370664885
--- FAIL: TestBackupRestoreSystemJobs (380.49s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=30</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestoreSystemJobs.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34865

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -1229,7 +1229,6 @@ func TestBackupRestoreSystemJobs(t *testing.T) {
 	backupTableID := sqlutils.QueryTableID(t, conn, "data", "public", "bank")
 
 	sqlDB.Exec(t, `CREATE DATABASE restoredb`)
-	restoreDatabaseID := sqlutils.QueryDatabaseID(t, conn, "restoredb")
 
 	// We create a full backup so that, below, we can test that incremental
 	// backups sanitize credentials in "INCREMENTAL FROM" URLs.
@@ -1265,10 +1264,6 @@ func TestBackupRestoreSystemJobs(t *testing.T) {
 			`RESTORE TABLE bank FROM '%s', '%s' WITH OPTIONS (into_db = 'restoredb')`,
 			sanitizedFullDir+"redacted", sanitizedIncDir+"redacted",
 		),
-		DescriptorIDs: descpb.IDs{
-			descpb.ID(restoreDatabaseID + 1),
-			descpb.ID(restoreDatabaseID + 2),
-		},
 	}); err != nil {
 		t.Fatal(err)
 	}

--- a/pkg/testutils/jobutils/jobs_verification.go
+++ b/pkg/testutils/jobutils/jobs_verification.go
@@ -159,11 +159,13 @@ func verifySystemJob(
 		runningStatusString = runningStatus.String
 	}
 
-	for _, id := range rawDescriptorIDs {
-		actual.DescriptorIDs = append(actual.DescriptorIDs, descpb.ID(id))
+	if len(expected.DescriptorIDs) > 0 {
+		for _, id := range rawDescriptorIDs {
+			actual.DescriptorIDs = append(actual.DescriptorIDs, descpb.ID(id))
+		}
+		sort.Sort(actual.DescriptorIDs)
+		sort.Sort(expected.DescriptorIDs)
 	}
-	sort.Sort(actual.DescriptorIDs)
-	sort.Sort(expected.DescriptorIDs)
 	expected.Details = nil
 	if e, a := expected, actual; !assert.Equal(logT{t}, e, a) {
 		return errors.Errorf("job %d did not match:\n%s",

